librecad/src/lib/math/rs_math.cpp

```diff
@@ -239,11 +239,93 @@ bool RS_Math::isSameDirection(double dir1, double dir2, double tol) {
 	return getAngleDifferenceU(dir1, dir2) < tol;
 }
 
+
+/*
+    Parse for units in expression, and then convert their values into millimeter.
+
+    - by Melwyn Francis Carlo.
+*/
```

Comment:
**My primary contribution!** It takes in any units in a given expression, and converts its preceding numerical value, and then discard the units. It can also deal with extraneous spaces in the expression.